app/services/credits/progressive_billing_service.rb

```diff
@@ -58,17 +58,16 @@ def call
 
     attr_reader :invoice
 
-    def apply_credit_to_fees(credit, total_charges_amount)
-      base_amount = total_charges_amount - invoice.progressive_billing_credit_amount_cents
+    def apply_credit_to_fees(progressive_billing_invoice)
+      progressive_billing_invoice.fees.charge.each do |progressive_fee|
+        fee = invoice.fees.find_by(
+          charge_id: progressive_fee.charge_id,
+          charge_filter_id: progressive_fee.charge_filter_id,
+          grouped_by: progressive_fee.grouped_by
+        )
```

Comment:
For performance purpose, maybe you should first preload the list of fees of the invoice and use `find` on the list to retrieve the matching fee